### PR TITLE
Bring fixes from 3.2 line to 4.0

### LINKED ISSUE #1169: StreamConfigurationProviders cannot be loaded more than once
## Describe the bug
Using SpringBootEnvProvider in conjunction with PlaceholderResolver results in an error: 
```
Unhandled exception. System.InvalidOperationException: StreamConfigurationProviders cannot be loaded more than once.
   at Microsoft.Extensions.Configuration.StreamConfigurationProvider.Load()
   at Steeltoe.Extensions.Configuration.SpringBoot.SpringBootEnvProvider.Load()
```

## Steps to reproduce 
1.AddPlaceholderResolver
2.Use any extension method that adds AddSpringBootFromEnvironmentVariable 

## Expected behavior
Should work without throwing errors

## Environment (please complete the following information):
 - Steeltoe Version [e.g. 3.2.3]
 - .NET Version [e.g. .NET 6] 
 - Any other library versions to note

### LINKED ISSUE #1178: Setting the "Spring:Cloud:Config:Timeout" configuration property doesn't affect the HttpClient timeout
## Describe the bug
Setting the "Spring:Cloud:Config:Timeout" configuration property doesn't seem to work. The library seems to use the default of 6 seconds (although I see this will soon be changed to 60s: https://github.com/SteeltoeOSS/Steeltoe/pull/1060) regardless of what the timeout config property is set to. If the config server takes longer than 6s to respond, it results in an exception like the following:

```
fail: Steeltoe.Extensions.Configuration.ConfigServer.ConfigServerConfigurationProvider[0]
      Config Server exception: System.Threading.Tasks.TaskCanceledException: The request was canceled due to the configured HttpClient.Timeout of 6 seconds elapsing.
       ---> System.TimeoutException: The operation was canceled.
       ---> System.Threading.Tasks.TaskCanceledException: The operation was canceled.
       ---> System.IO.IOException: Unable to read data from the transport connection: The I/O operation has been aborted because of either a thread exit or an application request..
       ---> System.Net.Sockets.SocketException (995): The I/O operation has been aborted because of either a thread exit or an application request.
         --- End of inner exception stack trace ---
         at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
         at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource<System.Int32>.GetResult(Int16 token)
         at System.Net.Http.HttpConnection.InitialFillAsync(Boolean async)
         at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
         --- End of inner exception stack trace ---
         at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
         at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
         at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
         at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, Cance
llationToken originalCancellationToken)
         --- End of inner exception stack trace ---
         --- End of inner exception stack trace ---
         at System.Net.Http.HttpClient.HandleFailure(Exception e, Boolean telemetryStarted, HttpResponseMessage response, CancellationTokenSource cts, CancellationToken cancellationToken, CancellationTokenSource pendingRequestsC
ts)
         at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, Cance
llationToken originalCancellationToken)
         at Steeltoe.Extensions.Configuration.ConfigServer.ConfigServerConfigurationProvider.RemoteLoadAsync(String requestUri), path: http%3A%2F%2Flocalhost%3A8763%2Ffoo%2FProduction
```

I got the above exception even when my appsettings.json looks like this:
```json
{
  "$schema": "https://steeltoe.io/schema/latest/schema.json",
  "Spring": {
    "Application": {
      "Name": "foo"
    },
    "Cloud": {
      "Config": {
        "HealthEnabled": true,
        "DiscoveryEnabled": true,
        "Uri": "http://localhost:8763",
        "ValidateCertificates": false,
        "Timeout": 20000
      }
    }
  },
  "Logging": {
    "IncludeScopes": false,
    "LogLevel": {
      "Default": "Warning"
    }
  },
  "Management": {
    "Endpoints": {
      "Actuator": {
        "Exposure": {
          "Include": [ "*" ],
          "Exclude": [ "env", "refresh" ]
        }
      }
    }
  }
  // Discovery-first
  //,"Eureka": {
  //  "Client": {
  //    "ServiceUrl": "http://localhost:8761/eureka/"
  //  }
  //}
}
```

## Steps to reproduce 
1. Start with the "Simple" configuration sample here: https://github.com/SteeltoeOSS/Samples/tree/main/Configuration/src/Simple
2. Change appsettings.json to add the Timeout property under the Spring:Cloud:Config section and set it to something other than 6000 (e.g. 20000).
3. Startup a dummy config server that is very slow to respond to requests. I just used a new .net web api project with a controller method that gets routed all requests and then does a Task.Delay:
``` c#
[ApiController]
[Route("{**catchAll}")]
public class WeatherForecastController : ControllerBase
{
	
	[Route("/{**catchAll}")]
	public async Task<IActionResult> Echo()
	{
		await Task.Delay(10000);
		return Ok();
	}
}
```
4. Change the Spring:Cloud:Config:Uri property in appsettings.json as required to point to the dummy config server.
5. Start the "Simple" configuration sample and see the exception in the console output mentioning the timeout of 6 seconds.

## Expected behavior
Expected result is that the timeout used by the HttpClient would match the configured Timeout value.

## Environment (please complete the following information):
 - Steeltoe Version: tried both 3.2.3 and 3.2.4
 - Platform: N/A
 - OS: Windows
 - .NET Version: 6.0

## Additional context or links
The issue may be due to the way _httpClient is initialized here:
https://github.com/SteeltoeOSS/Steeltoe/blob/3.2.3/src/Configuration/src/ConfigServerBase/ConfigServerConfigurationProvider.cs#L143
since the settings used are the "defaultSettings":
https://github.com/SteeltoeOSS/Steeltoe/blob/3.2.3/src/Configuration/src/ConfigServerBase/ConfigServerConfigurationProvider.cs#L122

### LINKED ISSUE #1199: Authentication fails when using service discovery on Azure Spring Apps Enterprise
## Describe the bug
App fails to register with discovery server secured with .p12 certificate

## Environment (please complete the following information):
 - Platform: Azure Spring Apps Enterprise
 - Found with Steeltoe 3.2.4, present in 4.0 line as well

## Additional context or links
Stack trace:
```
fail: Steeltoe.Discovery.Eureka.Transport.EurekaHttpClient[0]                                                                         
      RegisterAsync Failed, request was made to https://<instance>.svc.azuremicroservices.io/eureka/default/eureka/apps/<app>, retry: 0
      System.Net.Http.HttpRequestException: The SSL connection could not be established, see inner exception.                                    
       ---> System.Security.Authentication.AuthenticationException: Authentication failed, see inner exception.                                  
       ---> System.ComponentModel.Win32Exception (0x8009030E): No credentials are available in the security package                              
         at System.Net.SSPIWrapper.AcquireCredentialsHandle(ISSPIInterface secModule, String package, CredentialUse intent, SCH_CREDENTIALS* scc)
         at System.Net.Security.SslStreamPal.AcquireCredentialsHandle(CredentialUse credUsage, SCH_CREDENTIALS* secureCredential)
         at System.Net.Security.SslStreamPal.AcquireCredentialsHandleSchCredentials(SslStreamCertificateContext certificateContext, SslProtocols protocols, EncryptionPolicy policy, Boolean isServer)
         at System.Net.Security.SslStreamPal.AcquireCredentialsHandle(SslStreamCertificateContext certificateContext, SslProtocols protocols, EncryptionPolicy policy, Boolean isServer)
         --- End of inner exception stack trace ---
         at System.Net.Security.SslStreamPal.AcquireCredentialsHandle(SslStreamCertificateContext certificateContext, SslProtocols protocols, EncryptionPolicy policy, Boolean isServer)
         at System.Net.Security.SecureChannel.AcquireClientCredentials(Byte[]& thumbPrint)
         at System.Net.Security.SecureChannel.GenerateToken(ReadOnlySpan`1 inputBuffer, Byte[]& output)
         at System.Net.Security.SecureChannel.NextMessage(ReadOnlySpan`1 incomingBuffer)
         at System.Net.Security.SslStream.ProcessBlob(Int32 frameSize)
         at System.Net.Security.SslStream.ReceiveBlobAsync[TIOAdapter](TIOAdapter adapter)
         at System.Net.Security.SslStream.ForceAuthenticationAsync[TIOAdapter](TIOAdapter adapter, Boolean receiveFirst, Byte[] reAuthenticationData, Boolean isApm)
         at System.Net.Http.ConnectHelper.EstablishSslConnectionAsync(SslClientAuthenticationOptions sslOptions, HttpRequestMessage request, Boolean async, Stream stream, CancellationToken cancellationToken)
         --- End of inner exception stack trace ---
         at System.Net.Http.ConnectHelper.EstablishSslConnectionAsync(SslClientAuthenticationOptions sslOptions, HttpRequestMessage request, Boolean async, Stream stream, CancellationToken cancellationToken)
         at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
         at System.Net.Http.HttpConnectionPool.CreateHttp11ConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
         at System.Net.Http.HttpConnectionPool.AddHttp11ConnectionAsync(HttpRequestMessage request)
         at System.Threading.Tasks.TaskCompletionSourceWithCancellation`1.WaitWithCancellationAsync(CancellationToken cancellationToken)
         at System.Net.Http.HttpConnectionPool.GetHttp11ConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
         at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
         at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
         at Steeltoe.Common.Http.ClientCertificateHttpHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
         at Microsoft.Extensions.Http.Logging.LoggingHttpMessageHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
         at Microsoft.Extensions.Http.Logging.LoggingScopeHttpMessageHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
         at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, CancellationToken originalCancellationToken)
         at Steeltoe.Discovery.Eureka.Transport.EurekaHttpClient.RegisterAsyncInternal(InstanceInfo info)
```

### LINKED ISSUE #1217: Setting a non-zero PollingInterval with FailFast enabled on the Config Server Configuration Provider can cause the application to crash
## Describe the bug
When FailFast is enabled along with a non-zero PollingInterval, the application will crash if there is an exception thrown when polling the config server to refresh the configuration (e.g. due to a timeout). The console output looks like the following:

```
fail: Steeltoe.Extensions.Configuration.ConfigServer.ConfigServerConfigurationProvider[0]
      Config Server exception: System.Threading.Tasks.TaskCanceledException: The request was canceled due to the configured HttpClient.Timeout of 6 seconds elapsing.
       ---> System.TimeoutException: The operation was canceled.
       ---> System.Threading.Tasks.TaskCanceledException: The operation was canceled.
       ---> System.IO.IOException: Unable to read data from the transport connection: The I/O operation has been aborted because of either a thread exit or an application request..
       ---> System.Net.Sockets.SocketException (995): The I/O operation has been aborted because of either a thread exit or an application request.
         --- End of inner exception stack trace ---
         at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
         at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource<System.Int32>.GetResult(Int16 token)
         at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
         --- End of inner exception stack trace ---
         at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
         at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
         at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
         at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, Cance
llationToken originalCancellationToken)
         --- End of inner exception stack trace ---
         --- End of inner exception stack trace ---
         at System.Net.Http.HttpClient.HandleFailure(Exception e, Boolean telemetryStarted, HttpResponseMessage response, CancellationTokenSource cts, CancellationToken cancellationToken, CancellationTokenSource pendingRequestsC
ts)
         at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, Cance
llationToken originalCancellationToken)
         at Steeltoe.Extensions.Configuration.ConfigServer.ConfigServerConfigurationProvider.RemoteLoadAsync(String requestUri), path: http%3A%2F%2Flocalhost%3A8763%2Ffoo%2FProduction
warn: Steeltoe.Extensions.Configuration.ConfigServer.ConfigServerConfigurationProvider[0]
      Could not locate PropertySource: System.Threading.Tasks.TaskCanceledException: The request was canceled due to the configured HttpClient.Timeout of 6 seconds elapsing.
       ---> System.TimeoutException: The operation was canceled.
       ---> System.Threading.Tasks.TaskCanceledException: The operation was canceled.
       ---> System.IO.IOException: Unable to read data from the transport connection: The I/O operation has been aborted because of either a thread exit or an application request..
       ---> System.Net.Sockets.SocketException (995): The I/O operation has been aborted because of either a thread exit or an application request.
         --- End of inner exception stack trace ---
         at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
         at System.Net.Sockets.Socket.AwaitableSockeUnhandled exception. tAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource<System.Int32>.GetResult(Int16 token)
         at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
         --- End of inner exception stack trace ---
         at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
         at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
         at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
         at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, Cance
llationToken originalCancellationToken)
         --- End of inner exception stack trace ---
         --- End of inner exception stack trace ---
         at System.Net.Http.HttpClient.HandleFailure(Exception e, Boolean telemetryStarted, HttpResponseMessage response, CancellationTokenSource cts, CancellationToken cancellationToken, CancellationTokenSource pendingRequestsC
ts)
         at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposSteeltoe.Extensions.Configuration.ConfigServer.ConfigSe
rverException: Could not locate PropertySource, fail fast property is set, failing
 ---> System.Threading.Tasks.TaskCanceledException: The request was canceled due to the configured HttpClient.Timeout of 6 seconds elapsing.
 ---> System.TimeoutException: The operation was canceled.
 ---> System.Threading.Tasks.TaskCanceledException: The operation was canceled.
 ---> System.IO.IOException: Unable to read data from the transport connection: The I/O operation has been aborted because of either a thread exit or an application request..
 ---> System.Net.Sockets.SocketException (995): The I/O operation has been aborted because of either a thread exit or an application request.
   --- End of inner exception stack trace ---
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource<System.Int32>.GetResult(Int16 token)
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, Cancellatio
nToken originalCancellationToken)
   --- End of inner exception stack trace ---
   --- End of inner exception stack trace ---
   at System.Net.Http.HttpClient.HandleFailure(Exception e, Boolean telemetryStarted, HttpResponseMessage response, CancellationTokenSource cts, CancellationToken cancellationToken, CancellationTokenSource pendingRequestsCts)   
   at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, Cancellatio
nToken originalCancellationToken)
   at Steeltoe.Extensions.Configuration.ConfigServer.ConfigServerConfigurationProvider.RemoteLoadAsync(String requestUri)
   at Steeltoe.Extensions.Configuration.ConfigServer.ConfigServerConfigurationProvider.DoLoad(Boolean updateDictionary)
   --- End of inner exception stack trace ---
   at Steeltoe.Extensions.Configuration.ConfigServer.ConfigServerConfigurationProvider.DoLoad(Boolean updateDictionary)
   at Steeltoe.Extensions.Configuration.ConfigServer.ConfigServerConfigurationProvider.<OnSettingsChanged>b__27_1(Object _)
   at System.Threading.TimerQueueTimer.<>c.<.cctor>b__27_0(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.TimerQueueTimer.CallCallback(Boolean isThreadPool)
   at System.Threading.TimerQueueTimer.Fire(Boolean isThreadPool)
   at System.Threading.TimerQueue.FireNextTimers()
   at System.Threading.TimerQueue.AppDomainTimerCallback(Int32 id)
eCts, CancellationTokenSource pendingRequestsCts, CancellationToken originalCancellationToken)
         at Steeltoe.Extensions.Configuration.ConfigServer.ConfigServerConfigurationProvider.RemoteLoadAsync(String requestUri)
         at Steeltoe.Extensions.Configuration.ConfigServer.ConfigServerConfigurationProvider.DoLoad(Boolean updateDictionary)

Process finished with exit code -532,462,766.

```

## Steps to reproduce 
1. Start with the "Simple" configuration sample here:
https://github.com/SteeltoeOSS/Samples/tree/main/Configuration/src/Simple
2. Change appsettings.json to add the FailFast property under the Spring:Cloud:Config section and set it to true
3. Change appsettings.json to add the PollingInterval property under the Spring:Cloud:Config section and set it to "00:00:15"
4. Start up a test config server and wait for it to finish starting.
5. Start up the "Simple" configuration sample application and wait for it to finish starting.
6. Stop the test config server.
7. Wait until the next polling interval (up to 15 seconds) for the sample application to crash.

## Expected behavior
The refresh polling should not crash the application. If it is unsuccessful in loading the configuration it should probably just log a warning.

## Environment (please complete the following information):
 - Steeltoe Version: 3.2.5
 - Platform: N/A 
 - OS: Windows
 - .NET Version: 6.0

## Additional context or links
This seems due to an unhandled exception being thrown within the [callback of a System.Threading.Timer](https://github.com/SteeltoeOSS/Steeltoe/blob/release/3.2/src/Configuration/src/ConfigServerBase/ConfigServerConfigurationProvider.cs#L163), which crashes the application. When FailFast is true, [an exception may be thrown within the Load method](https://github.com/SteeltoeOSS/Steeltoe/blob/release/3.2/src/Configuration/src/ConfigServerBase/ConfigServerConfigurationProvider.cs#L364). I think the timer callback should wrap the call to the Load method in a try/catch and log any exception.

### LINKED ISSUE #1218: Disabling the config server client doesn't disable refresh polling
## Describe the bug
When the Enabled property is set to false with a non-zero PollingInterval, the refresh polling is not disabled as expected. There are still requests made to the config server every polling interval. If the config server is unreachable, the console output looks like the following every interval:
```
fail: Steeltoe.Extensions.Configuration.ConfigServer.ConfigServerConfigurationProvider[0]                                                   
      Config Server exception: System.Net.Http.HttpRequestException: No connection could be made because the target machine actively refused it. (localhost:8763)                          
       ---> System.Net.Sockets.SocketException (10061): No connection could be made because the target machine actively refused it.                                                        
         at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)                                                 
         at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)                                                 
         at System.Net.Sockets.Socket.<ConnectAsync>g__WaitForConnectWithCancellation|277_0(AwaitableSocketAsyncEventArgs saea, ValueTask connectTask, CancellationToken cancellationToken)
         at System.Net.Http.HttpConnectionPool.ConnectToTcpHostAsync(String host, Int32 port, HttpRequestMessage initialRequest, Boolean async, CancellationToken cancellationToken)
         --- End of inner exception stack trace ---
         at System.Net.Http.HttpConnectionPool.ConnectToTcpHostAsync(String host, Int32 port, HttpRequestMessage initialRequest, Boolean async, CancellationToken cancellationToken)
         at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
         at System.Net.Http.HttpConnectionPool.CreateHttp11ConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
         at System.Net.Http.HttpConnectionPool.AddHttp11ConnectionAsync(HttpRequestMessage request)
         at System.Threading.Tasks.TaskCompletionSourceWithCancellation`1.WaitWithCancellationAsync(CancellationToken cancellationToken)
         at System.Net.Http.HttpConnectionPool.GetHttp11ConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
         at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
         at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
         at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, Cance
llationToken originalCancellationToken)
         at Steeltoe.Extensions.Configuration.ConfigServer.ConfigServerConfigurationProvider.RemoteLoadAsync(String requestUri), path: http%3A%2F%2Flocalhost%3A8763%2Ffoo%2FProduction
warn: Steeltoe.Extensions.Configuration.ConfigServer.ConfigServerConfigurationProvider[0]
      Could not locate PropertySource: System.Net.Http.HttpRequestException: No connection could be made because the target machine actively refused it. (localhost:8763)
       ---> System.Net.Sockets.SocketException (10061): No connection could be made because the target machine actively refused it.
         at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
         at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)
         at System.Net.Sockets.Socket.<ConnectAsync>g__WaitForConnectWithCancellation|277_0(AwaitableSocketAsyncEventArgs saea, ValueTask connectTask, CancellationToken cancellationToken)
         at System.Net.Http.HttpConnectionPool.ConnectToTcpHostAsync(String host, Int32 port, HttpRequestMessage initialRequest, Boolean async, CancellationToken cancellationToken)
         --- End of inner exception stack trace ---
         at System.Net.Http.HttpConnectionPool.ConnectToTcpHostAsync(String host, Int32 port, HttpRequestMessage initialRequest, Boolean async, CancellationToken cancellationToken)
         at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
         at System.Net.Http.HttpConnectionPool.CreateHttp11ConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
         at System.Net.Http.HttpConnectionPool.AddHttp11ConnectionAsync(HttpRequestMessage request)
         at System.Threading.Tasks.TaskCompletionSourceWithCancellation`1.WaitWithCancellationAsync(CancellationToken cancellationToken)
         at System.Net.Http.HttpConnectionPool.GetHttp11ConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
         at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
         at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
         at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, Cance
llationToken originalCancellationToken)
         at Steeltoe.Extensions.Configuration.ConfigServer.ConfigServerConfigurationProvider.RemoteLoadAsync(String requestUri)
         at Steeltoe.Extensions.Configuration.ConfigServer.ConfigServerConfigurationProvider.DoLoad(Boolean updateDictionary)

```  

## Steps to reproduce 
1. Start with the "Simple" configuration sample here:
https://github.com/SteeltoeOSS/Samples/tree/main/Configuration/src/Simple
2. Change appsettings.json to add the Enabled property under the Spring:Cloud:Config section and set it to false
3. Change appsettings.json to add the PollingInterval property under the Spring:Cloud:Config section and set it to "00:00:15"
4. Make sure the config server at the configured Uri is stopped
5. Start the "Simple" configuration sample
6. See the exception in the console output every 15 seconds

## Expected behavior
Refresh polling should also be disabled when the Enabled property is set to false

## Environment (please complete the following information):
 - Steeltoe Version: 3.2.5
 - Platform: N/A 
 - OS: Windows
 - .NET Version: 6.0

### PR DESCRIPTION
**Pull request by TimHess**
## Description

Apply fixes from several PRs against 3.2 in to 4.0. commits were applied in this order: #1142, #1170, #1200, #1208, #1219

Fixes #1169
Fixes #1199 
Fixes #1178 
Fixes #1218 
Fixes #1217 

## Quality checklist

- [x] Your code complies with our [Coding Style](https://github.com/SteeltoeOSS/.github/blob/main/contributing-docs/contributing-code-style.md).
- [x] You've updated unit and/or integration tests for your change, where applicable.
- [ ] You've updated documentation for your change, where applicable.
      If your change affects other repositories, such as [Documentation](https://github.com/SteeltoeOSS/Documentation), [Samples](https://github.com/SteeltoeOSS/Samples) and/or [MainSite](https://github.com/SteeltoeOSS/MainSite), add linked PRs here.
- [x] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.
- [x] You've added required license files and/or file headers (explaining where the code came from with proper attribution), where code is copied from StackOverflow, a blog, or OSS.

### CODE REVIEW

File: src/Common/test/Common.Security.Test/CertificateRotationServiceTest.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+        using var serviceProvider = services.BuildServiceProvider();
+        var service = serviceProvider.GetRequiredService<ICertificateRotationService>();
+
+        service.Start();
+
+        var collection = personalCertStore.Certificates.Find(X509FindType.FindByIssuerName, "Diego Instance Identity Intermediate CA", false);
+        collection.Should().NotBeNull();
+
+#if NET6_0_OR_GREATER
```
Comment (bart-vmware) [inline src/Common/test/Common.Security.Test/CertificateRotationServiceTest.cs:48]: I suppose these can be removed?

File: src/Configuration/test/SpringBoot.Test/SpringBootHostBuilderExtensionsTest.cs [hunk 2]
```diff
@@ -75,6 +76,24 @@ public void GenericHostConfiguresIConfiguration_Spring_Application_Json()
         Assert.Equal("value", configuration["foo:bar"]);
     }
 
+    [Fact]
+    public void GenericHostConfiguresIConfiguration_Spring_Application_Json_Works_With_Placeholder()
+    {
+        Environment.SetEnvironmentVariable("SPRING_APPLICATION_JSON", "{\"foo.bar\":\"value\"}");
```
Comment (bart-vmware) [inline src/Configuration/test/SpringBoot.Test/SpringBootHostBuilderExtensionsTest.cs:82]: Use `EnvironmentVariableScope` instead.

File: src/Common/test/Common.Security.Test/CertificateRotationServiceTest.cs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    {
+        using var personalCertStore = new X509Store(StoreName.My, StoreLocation.CurrentUser);
+        personalCertStore.Open(OpenFlags.ReadWrite);
+        personalCertStore.Certificates.Find(X509FindType.FindByIssuerName, "Diego Instance Identity Intermediate CA", false).Should().BeEmpty();
+
+        using var sandbox = new Sandbox();
+        var filename = sandbox.CreateFile("fakeCertificate.p12");
+        File.Copy("instance.p12", filename, true);
+
+        var config = new ConfigurationBuilder()
```
Comment (bart-vmware) [inline src/Common/test/Common.Security.Test/CertificateRotationServiceTest.cs:31]: Please don't reintroduce abbreviations like "config", "cert", "certWriter" etc. in tests.

**Review by bart-vmware — CHANGES_REQUESTED** [review]

**Comment by bart-vmware** [comment]
Looks like `pr-code-cleanup.yml` is broken. To fix it, replace:
`--dotnetcoresdk=$(dotnet --version)`
with:
`--jb --dotnetcoresdk=$(dotnet --version)`

File: src/Common/test/Common.Security.Test/CertificateRotationServiceTest.cs (hunk 3 above)
Comment (TimHess) [reply src/Common/test/Common.Security.Test/CertificateRotationServiceTest.cs:31]: ah yeah, this is copy/pasted... I had already fixed a few but I'll take another pass and see if I can fix the rest

**Comment by TimHess** [comment]
> Looks like `pr-code-cleanup.yml` is broken. To fix it, replace: `--dotnetcoresdk=$(dotnet --version)` with: `--jb --dotnetcoresdk=$(dotnet --version)`

Thanks, I figured that out this morning from cleanupcode.ps1

**Review by bart-vmware — CHANGES_REQUESTED** [review]
Can you please add a commit to fix this one?

> Looks like pr-code-cleanup.yml is broken. To fix it, replace:
--dotnetcoresdk=$(dotnet --version)
with:
--jb --dotnetcoresdk=$(dotnet --version)

**Review by bart-vmware — APPROVED** [review]

**Pull request merged**